### PR TITLE
feat(OSC onboarding): New OSC form and validator

### PR DESCRIPTION
**Pull request by gustavlrsn**
Resolves https://github.com/opencollective/opencollective/issues/5974

# Description
- adds an interface `RepositoryInfo` with the information needed for OSC validation
- adds a function to fetch this information from GitHub
- adds a OSC validator function that checks this information and returns an object of type `ValidatedRepositoryInfo` with the values and whether they are valid according to the criteria
- adds the GitHub validation to the `applyToHost` mutation (already in `createCollective`), and adds the validated repository information to the applicationData in both of those mutations

### CODE REVIEW

**Comment by gustavlrsn** [comment]
@kewitz maybe something we want to fix or accept as tech debt:

The `createCollective` mutation has a `automateApprovalWithGithub` argument to trigger the automated GitHub approval.

The `applyToHost` mutation which now also supports automated GitHub approval does not have this argument, the resolver will instead check if the `applicationData.repositoryUrl` is a GitHub URL.

I think it could make sense to remove this argument completely from `createCollective` and just check if the repository is a GitHub one on the API side. What do you think?

**Comment by gustavlrsn** [comment]
Also something to consider, there is no OSC validation (except for checking that someone is an admin) when applying with only an organization account (and no repository). Currently we check whether at least one repository in the org has 100 stars. 

How to extend that to the new OSC validation might need some thinking. Maybe we should check all repositories, and see if at least one of them pass, and show that in the pending applications view?

**Comment by gustavlrsn** [comment]
Looking at the failing GraphQL test..

**Comment by gustavlrsn** [comment]
Test fixed now, needed to add a mocked response from the GitHub GraphQL API.

**Comment by BenJam** [comment]
> we should check all repositories, and see if at least one of them pass

that seems sensible to me, if you're able to do that without a call for each repo (which might take forever, for instance, applying with an account like opencollective's: 

Screenshot 2022-10-12 at 15 29 52

_Gulp!_

Also if someone is applying on behalf of an organisation i feel like it would be sensible to enable them to select >1 license from the dropdown. 

Finally, how are we dealing with the (optional) parts. The intent is that the user can apply (using the form) on behalf of an open source project or as an open source 'adjacent' community, at which point only the bottom part of the form is relevant. I know there was a design in which we had both parts of the form in two separate tabs, but we ended up with one continuous flow... it might be useful to talk with @aminakazi about this.

**Comment by gustavlrsn** [comment]
> > we should check all repositories, and see if at least one of them pass
> 
> that seems sensible to me, if you're able to do that without a call for each repo (which might take forever, for instance, applying with an account like opencollective's:
> 
> Screenshot 2022-10-12 at 15 29 52
> 
> _Gulp!_

In a call earlier with @znarf & @kewitz we decided to try to simplify the repo picker so that you can only choose a repository to do the validation with.

> Finally, how are we dealing with the (optional) parts. The intent is that the user can apply (using the form) on behalf of an open source project or as an open source 'adjacent' community, at which point only the bottom part of the form is relevant. I know there was a design in which we had both parts of the form in two separate tabs, but we ended up with one continuous flow... it might be useful to talk with @aminakazi about this.

I had missed to add the collapsable sections to the form, I've added those now at least, maybe improves things a little bit..

**Review by kewitz — APPROVED** [review]
Nice work, it looks simple and consistent.

File: server/controllers/connectedAccounts.js [hunk 1]
```diff
@@ -179,9 +179,13 @@ export const fetchAllRepositories = async (req, res, next) => {
     req.setTimeout(GITHUB_REPOS_FETCH_TIMEOUT);
     let repos = await github.getAllUserPublicRepos(githubAccount.token);
     if (repos.length !== 0) {
-      repos = repos.filter(repo => {
-        return repo.stargazers_count >= config.githubFlow.minNbStars && repo.fork === false;
-      });
+      repos = repos
+        .filter(repo => {
+          return repo.fork === false;
```
[the remark this answers is not in the document]
Comment (gustavlrsn) [reply server/controllers/connectedAccounts.js:184]: Yep, I removed the last FAQ question about 100 stars

**Comment by gustavlrsn** [comment]
Thanks Leo

**Pull request merged**